cmd/dep/init.go

```diff
@@ -105,39 +107,50 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	sm.UseDefaultSignalHandling()
 	defer sm.Release()
 
-	pd, err := getProjectData(ctx, pkgT, cpr, sm)
-	if err != nil {
-		return err
-	}
+	// Create empty manifest, lock and projectData, and fill them according to
+	// init operation modes. If gopath flag is set, fetch projectData by searching
+	// through GOPATH, else operate in network mode, solve all the dependencies
+	// over network.
 	m := &dep.Manifest{
-		Dependencies: pd.constraints,
+		Dependencies: make(gps.ProjectConstraints),
 	}
+	l := &dep.Lock{}
+	var pd projectData
 
-	// Make an initial lock from what knowledge we've collected about the
-	// versions on disk
-	l := &dep.Lock{
-		P: make([]gps.LockedProject, 0, len(pd.ondisk)),
-	}
+	if cmd.gopath {
+		pd, err = getProjectData(ctx, pkgT, cpr, sm)
+		if err != nil {
+			return err
+		}
 
-	for pr, v := range pd.ondisk {
-		// That we have to chop off these path prefixes is a symptom of
-		// a problem in gps itself
-		pkgs := make([]string, 0, len(pd.dependencies[pr]))
-		prslash := string(pr) + "/"
-		for _, pkg := range pd.dependencies[pr] {
-			if pkg == string(pr) {
-				pkgs = append(pkgs, ".")
-			} else {
-				pkgs = append(pkgs, trimPathPrefix(pkg, prslash))
-			}
+		m.Dependencies = pd.constraints
+
+		// Make an initial lock from what knowledge we've collected about the
+		// versions on disk
+		l = &dep.Lock{
+			P: make([]gps.LockedProject, 0, len(pd.ondisk)),
 		}
 
-		l.P = append(l.P, gps.NewLockedProject(
-			gps.ProjectIdentifier{ProjectRoot: pr}, v, pkgs),
-		)
+		for pr, v := range pd.ondisk {
+			// That we have to chop off these path prefixes is a symptom of
+			// a problem in gps itself
```

Comment:
> With little details about how to create `LockedProject` with proper list of packages using gps

Well, this is kinda what I'm getting at - the "properness" of the package list in `LockedProject`'s isn't separable from a full `Solve()`. [`NewLockedProject()`](https://godoc.org/github.com/sdboyer/gps#NewLockedProject) will let you create instances as needed, but being proper/correct is only defined in the context of an overall dependency set, computed by solving.

So, is the question here about simply creating instances, or creating _correct_ ones?